apps/web/src/pages/workspace/[workspaceId]/[pageId].tsx

```diff
@@ -124,6 +125,16 @@ const PageDefender = ({ children }: PropsWithChildren) => {
     };
     initPage();
   }, [createPage, currentWorkspace, loadPage, router.query.pageId]);
+  const api = useGlobalStateApi();
+  useEffect(
+    () =>
+      dataCenter.onWorkspacesChange(({ deleted }) => {
+        if (deleted?.some(workspace => workspace.id === currentWorkspace?.id)) {
+          router.replace('/404?code=kicked');
+        }
+      }),
+    [api, currentWorkspace?.id, dataCenter, router]
+  );
```

Comment:
There is some similar 404 handling in `useEnsureWorkspace`. Do you think it is better handled there?

Reply:
We could do this later